framework/include/utils/MooseUtils.h

```diff
@@ -242,6 +233,54 @@ std::string stripExtension(const std::string & s);
  */
 std::pair<std::string, std::string> splitFileName(std::string full_file);
 
+std::string pathjoin(const std::string & s);
+
+/**
+ * Join parts of a path together to form the full path
+ * Note: Paths such as '.' or '..' will be removed properly by this function.
+ */
+template <typename... Args>
+std::string
+pathjoin(const std::string & s, Args... args)
+{
+  std::string path = s;
+  if (path[path.size() - 1] == '/')
+    path = path.substr(0, path.size() - 1);
+
+  std::string rest = pathjoin(args...);
+  if (rest.size() >= 2 && rest.substr(0, 2) == "..")
+  {
+    path = splitFileName(path).first;
+    if (rest.size() > 2)
+    {
+      if (rest[2] != '/')
+        mooseError("Back slash sign must be followed by '..'");
+      rest = rest.substr(3);
+      if (!rest.empty())
+        path = path + "/" + rest;
+    }
+  }
+  else if (rest.size() >= 1 && rest[0] == '.')
+  {
+    if (rest.size() > 1)
+    {
+      if (rest[1] != '/')
+        mooseError("Back slash sign must be followed by '.'");
+      rest = rest.substr(2);
+      if (!rest.empty())
+        path = path + "/" + rest;
+    }
+  }
+  else if (rest.size() > 0)
+  {
+    if (rest[0] == '/')
+      mooseError("Cannot join an absolute path");
+    path = path + "/" + rest;
+  }
+
+  return path;
+}
```

Comment:
this is from your other PR. I think it'd be best to keep those issues separated.

Reply:
You mean from the other issue? I do not need this commit and can drop it. But as I commented in the issue, if your PR cannot be merged soon, I rather this gets merged first. The screen printout is just pure stupid.

Reply:
Oh, saw your comment on the issue. So we agree to have a quick fix. I think unless we need more unit tests on this function, we can ship it with this. Our test suites can make sure no mistakes in this function.